ethereum_history_api/tests/src/get_header.e2e.test.ts

```diff
@@ -0,0 +1,53 @@
+import { describe, expect, it } from 'vitest';
+import { copy, updateNestedField, incHexStr } from 'noir-ethereum-api-oracles';
+import { abiEncode, WitnessMap } from '@noir-lang/noirc_abi';
+
+import { privateKeyToAccount } from 'viem/accounts';
+import getHeaderVerifier from '../../contracts/out/GetHeaderUltraPLONKVerifier.sol/UltraVerifier.json';
+import { Hash } from 'viem';
+import { createAnvilClient } from './ethereum/anvilClient.js';
+import { get_header_circuit, readInputMap, readProof, readWitnessMap } from './main.js';
+import {
+  ANVIL_TEST_ACCOUNT_PRIVATE_KEY,
+  FoundryArtefact,
+  VERIFICATION_GAS_LIMIT,
+  deployVerificationContract,
+  verifyStorageProofInSolidity
+} from './solidityVerifier.js';
+
+describe('get_header', async () => {
+  const account = privateKeyToAccount(ANVIL_TEST_ACCOUNT_PRIVATE_KEY);
+  const client = createAnvilClient();
+  const PROOF_PATH = '../../proofs/get_header.proof';
+  const INPUT_MAP_PATH = '../circuits/get_header/Verifier.toml';
+  const proof = await readProof(PROOF_PATH);
+  const inputMap = await readInputMap(INPUT_MAP_PATH);
+  const witnessMap = await readWitnessMap(INPUT_MAP_PATH, get_header_circuit.abi);
+  const verifierArtefact = getHeaderVerifier as FoundryArtefact;
+  const contractAddress = await deployVerificationContract(client, account, verifierArtefact);
+
+  function verifyProof(witnessMap: WitnessMap): Promise<Hash> {
+    return verifyStorageProofInSolidity(client, account, contractAddress, verifierArtefact.abi, proof, witnessMap);
```

Comment:
Both `verifyProof` and `verifyStorageProofInSolidity` calls one other function - we could merge them